### PR TITLE
Fixed afterChange behaviour and implemented beforeChange

### PR DESCRIPTION
**Pull request by vramana**

### CODE REVIEW

**Comment by vramana** [comment]
Fixes #56

File: src/mixins/helpers.js [hunk 1]
```diff
@@ -190,17 +190,27 @@ var helpers = {
...
+      this.setState(nextStateChanges);
+    };
+
+    var callback2 = () => {
+      this.setState(nextStateChanges);
+
+      if (this.props.afterChange) {
+        this.props.afterChange(currentSlide);
+      }
+    };
```
Comment (vramana) [inline src/mixins/helpers.js]: @akiran Instead of callback1 and callback2. Can this be written more nicely with a single callback?

Comment (akiran) [inline src/mixins/helpers.js]: addEndEventListener, and setTimeout are for same purpose. We should not give different callback to them.
setTimeout is backup for addEndEventerListener.

What if you call the afterChange function in callback and pass the same callback to both. Will it work ?

File: src/mixins/helpers.js [hunk 2]
```diff
@@ -190,17 +190,27 @@ var helpers = {
...
+      swipeLeft: null
+    };
+
+    // var callback1 = () => {
+    //   this.setState(nextStateChanges);
+    // };
+
+    var callback2 = () => {
+      this.setState(nextStateChanges);
+
```
Comment (akiran) [inline src/mixins/helpers.js]: You can clean up this next PR.
Just call it callback

**Pull request merged**